### PR TITLE
feat: Add `explain analyze graphical` feature for SQL performance visualization

### PR DESCRIPTION
**Pull request by Maricaya**
- Implemented a new `explain analyze graphical` command to visualize SQL performance analysis.
- Integrated front-end graphical components to display detailed query execution plans.
This update provides an intuitive interface for users to analyze and optimize SQL queries visually.

https://github.com/user-attachments/assets/fe276f03-a176-43a2-9b9a-8097201dda12

### CODE REVIEW

**Comment by Chasen-Zhang** [comment]
Please remove the "✨" in the title

**Comment by Chasen-Zhang** [comment]
The contributor is from the Summer of Open Source Promotion Plan. Please help with the review.  @PsiACE @zhang2014 @everpcpc

File: frontend/src/utills/index.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const percent = (numerator / denominator) * 100;
+  return `${percent.toFixed(1)}%`;
+}
+
+/**
+ * Formats a given millisecond value into a human-readable time format.
+ * @param {number} milliseconds - The value in milliseconds to be formatted.
+ * @returns {string} - The formatted time string.
+ */
+export function filterMillisecond(milliseconds: number): string {
```
Comment (Chasen-Zhang) [inline frontend/src/utills/index.ts:22]: This function is too complicated to write， Please use the package: pretty-ms

https://www.npmjs.com/package/pretty-ms

**Review by Chasen-Zhang — APPROVED** [review]
LGTM

**Comment by Chasen-Zhang** [comment]
Please resolve the conflict first @Maricaya

**Comment by Chasen-Zhang** [comment]
@sundy-li Request to merge into the main branch, thanks!

**Comment by sundy-li** [comment]
> @sundy-li Request to merge into the main branch, thanks!

do we need to wait for another pr in databend？

**Comment by Chasen-Zhang** [comment]
> > @sundy-li Request to merge into the main branch, thanks!
> 
> do we need to wait for another pr in databend？
> 
> @Maricaya please add `frontend/**` to licenserc.toml to fix the license check.

Let's wait for the databend pr to merge into the main branch

**Comment by Maricaya** [comment]
> > > @sundy-li Request to merge into the main branch, thanks!
> > 
> > 
> > do we need to wait for another pr in databend？
> > @Maricaya please add `frontend/**` to licenserc.toml to fix the license check.
> 
> Let's wait for the databend pr to merge into the main branch

got it!

**Comment by Chasen-Zhang** [comment]
@sundy-li This PR https://github.com/databendlabs/databend/pull/16543 is already in the main branch.

**Comment by Maricaya** [comment]
@sundy-li  this PR is ready, would you mind merging it？

**Comment by sundy-li** [comment]
Ok, It will be merged after ci pass.

**Comment by sundy-li** [comment]
@Maricaya Thanks for the contribution.

Some comments about this pr.

I checked the codes and looks like it will start a new server to listen to a free port in each `graphical query`, this may cause too much port resource usage.

1. Can we start the HTTP server as a service if it's `is_repl` is true?
2. If we receive `graphical query` in `is_repl` mode, we can generate a random URL to render the data for this query.

**Comment by PsiACE** [comment]
cool, glad to see this project has landed, thanks @Maricaya @Chasen-Zhang @sundy-li

**Comment by Maricaya** [comment]
> @Maricaya Thanks for the contribution.
> 
> Some comments about this pr.
> 
> I checked the codes and looks like it will start a new server to listen to a free port in each `graphical query`, this may cause too much port resource usage.
> 
> 1. Can we start a dedicated HTTP server as a service if it's `is_repl` is true?
> 2. If we receive `graphical query` in `is_repl` mode, we can generate a random URL to render the data for this query.

Got it, thanks for the feedback. Let me finish the CI process first. Here's the PR: https://github.com/databendlabs/bendsql/pull/486. I'll let you know once it's ready.

**Pull request merged**